pkg/sql/jobs_profiler_bundle.go

```diff
@@ -154,6 +155,35 @@ func (e *ExecutionDetailsBuilder) ReadExecutionDetail(
 	return buf.Bytes(), nil
 }
 
+// ListExecutionDetailFiles lists all the files that have been generated as part
+// of a job's execution details.
+func (e *ExecutionDetailsBuilder) ListExecutionDetailFiles(ctx context.Context) ([]string, error) {
+	var res []string
+	if err := e.db.Txn(ctx, func(ctx context.Context, txn isql.Txn) error {
+		jobInfo := jobs.InfoStorageForJob(txn, e.jobID)
+
+		// Iterate over all the files that have been stored as part of the job's
+		// execution details.
+		files := make([]string, 0)
+		if err := jobInfo.Iterate(ctx, profilerconstants.ExecutionDetailsChunkKeyPrefix,
+			func(infoKey string, value []byte) error {
+				// Look for the final chunk of each file to find the unique file name.
+				if idx := strings.Index(infoKey, finalChunkSuffix); idx != -1 {
```

Comment:
This relies on someone not putting _final in their name, right? If we changed `MakeProfilerBundleChunkKey` to take a _string_ chunk name, we could use the sentinel _as the chunk_ and then this logic would be strings.HasSuffix/strings.TrimSuffix, since the sentinel would be in a well-known place

Reply:
changed to this approach